cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -72,6 +72,7 @@ set(ARROW_THIRDPARTY_DEPENDENCIES
     Snappy
     Substrait
     Thrift
+    Ucx
```

How about using `ucx` because UCX uses `ucx` for CMake package name https://github.com/openucx/ucx/blob/master/cmake/ucx-targets.cmake.in ?

```suggestion
    ucx
```